File: JAVA/TypingSpeed.java
import java.time.Instant;
import java.time.Duration;
import java.util.Scanner;

public class TypingSpeed{
public static void main(String[] args){

Scanner input = new Scanner(System.in);

String sentenceToBeDisplayed = "The program should display a sentence for the user.";
Instant startTime = Instant.now();
System.out.println(sentenceToBeDisplayed);
String userResponse = input.nextLine();
Instant endTime = Instant.now();

Duration duration = Duration.between(startTime, endTime);
long durationInSeconds = duration.getSeconds();

long durationInMinutes = duration.toMinutes();
System.out.println(durationInMinutes);


int numberOfWords = countWordsInSentence(userResponse);
long wordsPerMinute = numberOfWords / durationInMinutes;



int numberOfErrors = countErrors(sentenceToBeDisplayed, userResponse);
int totalKeyStrokes = userResponse.length();

int correctKeyStrokes = totalKeyStrokes - numberOfErrors;



int accuracyPercentage = (correctKeyStrokes / totalKeyStrokes) * 100;

System.out.printf("Your result is stated below:%nTime taken (seconds): %d%nWords per minute: %d%nAccuracy percentage: %d%n", durationInSeconds, wordsPerMinute, accuracyPercentage);
}





public static int countWordsInSentence(String str){
	String[] wordsInArray = str.split(" ");
	int count = 0;
	for(int index = 0; index < wordsInArray.length; index++){
		count++;
	}
	return count;
}

public static int countErrors(String systemStr, String userResponse){
	int count = 0;
	String largest = userResponse;
	if (systemStr.length() >  userResponse.length()){
		largest = systemStr;
	}
	for(int index = 0; index < userResponse.length(); index++){
		if (userResponse.charAt(index) != systemStr.charAt(index))
			count++;
		if ((systemStr.length() - 1) == index)
			break;
	}
	System.out.println(count);
	int minuend = systemStr.length();
	System.out.println(minuend);
	int subtrahend = userResponse.length();
	System.out.println(subtrahend);
	int lengthDifference = 0;
	if (systemStr.length() != userResponse.length()){
		if (systemStr.length() <  userResponse.length()){
			minuend = userResponse.length();
			subtrahend = systemStr.length();
		}
		lengthDifference = minuend - subtrahend;
		System.out.println(lengthDifference);
	}
	count = count + lengthDifference;
	return count;
}














}

File: Python/typing_speed.py
import time

def count_words_in_sentence(str):
	words_in_list = str.split(" ")
	count = 0;
	for element in words_in_list:
		count += 1
	return count


def count_errors(systemStr, userResponse):
	count = 0;
	for index in range(0, len(userResponse)):
		if userResponse[index] != systemStr[index]:
			count += 1
			
		if len(systemStr) - 1 == index:
			break
	
	print(count)
	minuend = len(systemStr)
	print(minuend)
	subtrahend = len(userResponse)
	print(subtrahend);
	lengthDifference = 0;
	if len(systemStr) != len(userResponse):
		if len(systemStr) <  len(userResponse):
			minuend = len(userResponse)
			subtrahend = len(systemStr)
		lengthDifference = minuend - subtrahend
		print(lengthDifference)
	count = count + lengthDifference;
	return count





sentence_to_be_displayed = "The program should display a sentence for the user.";
start_time = time.time()
print(sentence_to_be_displayed)
user_response = input("")
end_time = time.time()

duration_in_seconds = end_time - start_time

duration_in_minutes = (int) duration_in_seconds // 60
print(duration_in_minutes)


number_of_words = count_words_in_sentence(user_response)
words_per_minute = (int) number_of_words // duration_in_minutes

number_of_errors = count_errors(sentence_to_be_displayed, user_response);
total_key_strokes = len(user_response)

correct_key_strokes = total_key_strokes - number_of_errors

accuracy_percentage = (correct_key_strokes / total_key_strokes) * 100;

print(f"Your result is stated below:\nTime taken (seconds): {duration_in_seconds}\nWords per minute: {words_per_minute}\nAccuracy percentage: {accuracy_percentage}\n")






